pkg/controller/storagecluster/cephcluster_test.go

```diff
@@ -355,3 +359,82 @@ func TestStorageClassDeviceSetCreation(t *testing.T) {
 	}
 
 }
+
+func createDummyKMSConfigMap(kmsProvider, kmsAddr string) *corev1.ConfigMap {
+	cm := &corev1.ConfigMap{}
+	cm.Name = KMSConfigMapName
+	cm.Data = make(map[string]string)
+	cm.Data["KMS_PROVIDER"] = kmsProvider
+	cm.Data[kmsProviderAddressKeyMap[kmsProvider]] = kmsAddr
+	cm.Data["VAULT_BACKEND_PATH"] = "ocs"
+	cm.Data["VAULT_NAMESPACE"] = "my-ocs-namespace"
+	return cm
+}
+
+func TestKMSConfigChanges(t *testing.T) {
+	ctxTodo := context.TODO()
+	cr := &api.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ocsinit",
+		},
+		Spec: api.StorageClusterSpec{
+			Encryption: api.EncryptionSpec{
+				KeyManagementService: api.KeyManagementServiceSpec{
+					Enable: true,
+				},
+			},
+		},
+	}
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      generateNameForCephCluster(cr),
+			Namespace: "",
+		},
+	}
+	validKMSArgs := []struct {
+		label       string
+		kmsProvider string
+		kmsAddress  string
+	}{
+		{label: "case 1", kmsProvider: "vault", kmsAddress: "http://localhost:5050"},
+		{label: "case 2", kmsProvider: "vault", kmsAddress: "http://localhost:12321"},
+		// ocs-operator is agnostic to KMS Provider, here rook should be throwing error
+		{label: "case 3", kmsProvider: "newKMSProvider", kmsAddress: "http://127.0.0.1:1553"},
+		// invalid test cases, make sure label has a prefix 'invalid'
+		{label: "invalid-case 1", kmsProvider: "vault", kmsAddress: "http://unearchable.url.location:3366"},
+	}
+	for _, kmsArgs := range validKMSArgs {
+		kmsCM := createDummyKMSConfigMap(kmsArgs.kmsProvider, kmsArgs.kmsAddress)
+		reconciler := createFakeInitializationStorageClusterReconciler(t, &nbv1.NooBaa{})
+		if err := reconciler.client.Create(ctxTodo, kmsCM); err != nil {
+			t.Errorf("Unable to create KMS configmap: %v", err)
+			t.FailNow()
+		}
+		// don't start dummy servers for invalid tests
+		if !strings.HasPrefix(kmsArgs.label, "invalid") {
+			startServerAt(kmsArgs.kmsAddress)
+		}
+		// have to initialize the image status,
+		// without which the code will throw a 'nil pointer' exception
+		reconciler.initializeImagesStatus(cr)
+		err := reconciler.ensureCephCluster(cr, reconciler.reqLogger)
+		if strings.HasPrefix(kmsArgs.label, "invalid") {
+			assert.Error(t, err, fmt.Sprintf("Failed: %q. Expected an error", kmsArgs.label))
+		} else {
+			assert.NoError(t, err, fmt.Sprintf("Failed: %q. Not expecting any error. Recieved: %v", kmsArgs.label, err))
+		}
+		// following part of the tests are only for valid tsets
+		if !strings.HasPrefix(kmsArgs.label, "invalid") {
```

Comment:
You can just remove these three lines and have the valid test path in the same stanza.

Reply:
Done...